skactiveml/pool/_core_set.py

```diff
@@ -0,0 +1,309 @@
+       (`candidates = X[mapping]`)
+    n_new_cand: int or None (default=None)
+       The number of new candidates that are additionally added to X.
+       Only used for the case, that in the query function with the
+       shape of candidates is (n_candidates, n_feature)
+
+    Return:
+    ----------
+    query_indices: numpy.ndarry of shape (batch_size, )
+        The query_indices indicate for which candidate sample a label is
+        to queried from the candidates.
+        If candidates in None or of shape (n_candidates), the indexing
+        refers to samples in X.
+        If candidates is of shape (n_candidates, n_features), the indexing
+        refers to samples in candidates.
+    utilities: numpy.ndarray of shape (batch_size, n_samples) or
+        numpy.ndarry of shape (batch_size, n_new_cand)
+        The distance between each data point and its nearest center that used
+        for selecting the next sample.
+        If candidates is None or of shape (n_candidates), the indexing
+        refers to samples in X.
+        If candidates is of shape (n_candidates, n_features), the indexing
+        refers to samples in candidates.
+    """
+
+    # valid the input shape whether is valid or not.
+    X = check_array(X, allow_nd=True)
+    y = check_array(
+        y, ensure_2d=False, force_all_finite="allow-nan", dtype=None
+    )
+    y = column_or_1d(y, warn=True)
+    check_consistent_length(X, y)
+
+    selected_samples = labeled_indices(y, missing_label=missing_label)
+
+    random_state_ = check_random_state(random_state)
+
+    if mapping is None:
+        mapping = unlabeled_indices(y, missing_label=missing_label)
+    else:
+        mapping = column_or_1d(mapping, dtype=int, warn=True)
+
+    if not isinstance(batch_size, int):
+        raise TypeError("batch_size must be a integer")
+
+    # initialize the utilities matrix with
+    if n_new_cand is None:
+        utilities = np.empty(shape=(batch_size, X.shape[0]))
+    elif isinstance(n_new_cand, int):
+        if n_new_cand == len(mapping):
+            utilities = np.empty(shape=(batch_size, n_new_cand))
+        else:
+            raise ValueError("n_new_cand must equal to the length of mapping array")
+    else:
+        raise TypeError("Only n_new_cand with type int is supported.")
+
+    query_indices = np.array([], dtype=int)
+
+    for i in range(batch_size):
+        if i == 0:
+            update_dist = _update_distances(X, selected_samples, mapping)
+        else:
+            latest_dist = utilities[i - 1]
+            update_dist = _update_distances(X, [idx], mapping, latest_dist)
+
+        if n_new_cand is None:
+            utilities[i] = update_dist
+        else:
+            utilities[i] = update_dist[mapping]
+
+        # select index
+        idx = np.nanargmax(utilities[i])
+
+        if len(selected_samples) == 0:
+            idx = random_state_.choice(mapping)
+            # because np.nanargmax always return the first occurrence is returned
```

Comment:
```suggestion
        idx = rand_argmax(utilities[i], random_state=random_state_)
```
cf. https://github.com/scikit-activeml/scikit-activeml/blob/master/skactiveml/utils/_selection.py